addons/web/static/src/core/color_picker/color_picker.xml

```diff
@@ -89,16 +89,21 @@
                     </t>
                     <button class="o_color_button btn selected" t-att-data-color="this.state.currentCustomColor" t-attf-style="background-color: {{this.state.currentCustomColor}}"/>
                 </div>
-                <div class="o_colorpicker_section" t-on-click="onColorApply" t-on-mouseover="onColorHover" t-on-mouseout="onColorHoverOut" t-on-focusin="onColorFocusin">
-                    <button data-color="black" class="btn o_color_button bg-black"></button>
-                    <button data-color="900" class="o_color_button btn" style="background-color: var(--900)"></button>
-                    <button data-color="800" class="o_color_button btn" style="background-color: var(--800)"></button>
-                    <button data-color="600" class="o_color_button btn" style="background-color: var(--600)"></button>
-                    <button data-color="400" class="o_color_button btn" style="background-color: var(--400)"></button>
-                    <button data-color="200" class="o_color_button btn" style="background-color: var(--200)"></button>
-                    <button data-color="100" class="o_color_button btn" style="background-color: var(--100)"></button>
-                    <button data-color="white" class="o_color_button btn bg-white"></button>
-                </div>
+                <t t-if="this.props.enabledGrayscales.includes('solid')" >
+                    <div class="o_colorpicker_section" t-on-click="onColorApply" t-on-mouseover="onColorHover" t-on-mouseout="onColorHoverOut" t-on-focusin="onColorFocusin">
+                        <t t-foreach="['black', '900', '800', '600', '400', '200', '100', 'white']" t-as="color" t-key="color">
+                            <button t-att-data-color="color" class="o_color_button btn" t-attf-style="background-color: var(--{{color}})" />
+                        </t>
+                    </div>
+                </t>
+                <t t-if="this.props.enabledGrayscales.includes('transparent')" >
+                    <div class="o_colorpicker_section" t-on-click="onColorApply" t-on-mouseover="onColorHover" t-on-mouseout="onColorHoverOut" t-on-focusin="onColorFocusin">
+                        <t t-set="transparentGrays" t-value="['black-75', 'black-50', 'black-25', 'black-15', 'white-25', 'white-50', 'white-75', 'white-85']" />
```

Comment:
why doing a `t-set` here and not in the previous `t-foreach` .. strange
but there's probably a way to avoid doing the same foreach twice here 😉 
uses a loop on both arrays (or nothing or only one) to create the `o_colorpicker_section`. This way, there's no duplication of code